homeassistant/components/notify/__init__.py

```diff
@@ -63,3 +76,60 @@ async def persistent_notification(service: ServiceCall) -> None:
     )
 
     return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a config entry."""
+    service_integration: ServiceIntegration = hass.data[DOMAIN]
+    return await service_integration.async_setup_entry(entry)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    service_integration: ServiceIntegration = hass.data[DOMAIN]
+    return await service_integration.async_unload_entry(entry)
+
+
+class NotifyService(PlatformService):
```

Comment:
It's now possible to completely customize the service in the platforms. Eg the telegram notify platform can register a service with a custom name, that takes custom service data parameters, validated by a custom service schema, with service descriptions loaded from the telegram integration.

On the one hand, this is good since it allows the platforms to tailor the service data validation according to their needs. On the other hand it makes the services possibly less standardized. Happy to hear more thoughts about this.

Reply:
One idea is to have a base service schema that only the service integration (`notify`) can define, and use this together with a possible custom service schema, defined by eg `telegram`, during service data validation. We then enforce the required service parameters that the service integration has defined while still allowing custom parameters by platforms.

We can also disallow overriding the service handler that handles the service call `async_handle_service` in the platforms.

Reply:
We’ve always allowed integrations to register any service they want under their own domain. We wouldn’t want telegram to register things under the ‘notify’ integration. 

I think that a base service schema is great, it is how we used to do `PLATFORM_SCHEMA` too.

Reply:
Currently in the legacy notify integration platforms can customize the name of the service registered. Do you mean that we should not allow this going forward?

It's when the platform calls `async_add_service` that the registration happens, so technically it *is* the platform that registers the service.